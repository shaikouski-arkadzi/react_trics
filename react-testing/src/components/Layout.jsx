import { useState, useEffect } from "react";
import List from "./List";
import Search from "./Search";

const initData = ["Text 1", "Text 2", "Text 3"];

const Layout = () => {
  const [searchValue, setSearchValue] = useState("");
  const [data, setData] = useState(initData);
  useEffect(() => {
    setData(
      initData.filter((el) =>
        el.toLowerCase().includes(searchValue.toLowerCase())
      )
    );
  }, [searchValue]);

  return (
    <>
      <Search
        value={searchValue}
        onChange={(e) => setSearchValue(e.target.value)}
      >
        Find:
      </Search>
      <List items={data} />
    </>
  );
};

export default Layout;
