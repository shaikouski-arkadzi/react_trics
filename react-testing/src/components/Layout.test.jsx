import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe } from "vitest";
import Layout from "./Layout";

describe("Layout component", () => {
  test("render Layout component", () => {
    expect(render(<Layout />)).toMatchSnapshot();
  });

  test("typing on Searchbox and filter works", async () => {
    render(<Layout />);
    await userEvent.type(screen.getByRole("textbox"), "Text 2");
    expect(screen.queryByDisplayValue(/Text 2/)).toBeInTheDocument();
    expect(screen.queryByDisplayValue(/Text 1/i)).toBeNull();
  });
});
