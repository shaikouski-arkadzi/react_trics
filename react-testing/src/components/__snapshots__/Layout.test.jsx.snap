// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout component > render Layout component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label>
        Find:
        <input
          type="text"
          value=""
        />
      </label>
      <ul>
        <li>
          Text 1
        </li>
        <li>
          Text 2
        </li>
        <li>
          Text 3
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <label>
      Find:
      <input
        type="text"
        value=""
      />
    </label>
    <ul>
      <li>
        Text 1
      </li>
      <li>
        Text 2
      </li>
      <li>
        Text 3
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
